### PR TITLE
​feat: new demo app

### PR DESCRIPTION
**Pull request by ShellyCoen**
This is the todo app that I have been working on that uses hapi.

### CODE REVIEW

File: demo/Public/index.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<!doctype HTML>
+
+<html>
```
Comment (Charlotteis) [inline demo/Public/index.html]: `<html lang="en">`

> "Declaring a language attribute on the html element enables a screen reader to read out the text with correct pronunciation." http://a11yproject.com/checklist.html

File: demo/Public/index.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+<head>
+  <title>You can do it!</title>
```
Comment (Charlotteis) [inline demo/Public/index.html]: Could we get a more descriptive title? It needs a little more context.
What about `Rocketship ToDo - You can do it!` though I am sure you can think of something else more magical :star2:

File: demo/Public/index.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  <meta charset="utf-8">
+  <link rel="stylesheet" type="text/css" href="style.css">
+
+</head>
+<body>
+
+  <center>
```
Comment (Charlotteis) [inline demo/Public/index.html]: I wouldn't use <center> like this. It is a deprecated (no longer suggested to use) tag. Better would be to add a container around the element and style it with a `margin: 0 auto` with CSS :D

File: demo/Public/index.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    <div id="logo-container"><img src="Rocket.png" /></div>
+
+    <form class="js-add-todo">
```
Comment (Charlotteis) [inline demo/Public/index.html]: The convention is to have JS hooks as IDs, rather than classes. ID's are for one time use :)

File: demo/Public/index.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      <div role="main" class="content">
+        <label for="todos"><h1>Let's get you organised!</h1></label>
```
Comment (Charlotteis) [inline demo/Public/index.html]: The label is for todos, but the label should be for the 'js-todo-task' input. Also the js-todo-task is better as an ID.

File: demo/Public/index.html [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        <input type="text" class="js-todo-task" placeholder="What do you need to do?"/>
+        <button type="submit">Add</button>
+      </div>
+      <p>
+        <button class="js-check-all">Check all</button>
```
Comment (Charlotteis) [inline demo/Public/index.html]: What about 'Check all items'?

File: demo/Public/index.html [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      </p>
+    </form>
+
+    <div>
+      <ul class="js-todo-container" style="list-style-type:none"></ul>
+      <a href="#" class="js-clear-todos"><h3>Clear Checked</h3></a>
```
Comment (Charlotteis) [inline demo/Public/index.html]: What about 'clear checked items'?

File: demo/Public/index.html [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        <input type="text" class="js-todo-task" placeholder="What do you need to do?"/>
+        <button type="submit">Add</button>
+      </div>
+      <p>
+        <button class="js-check-all">Check all</button>
+      </p>
+    </form>
+
+    <div>
+      <ul class="js-todo-container" style="list-style-type:none"></ul>
```
Comment (Charlotteis) [inline demo/Public/index.html]: I'd avoid inline styles as they can muddy the HTML, would it be better to add this in the CSS file instead?

File: demo/Public/index.html [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      <a href="#" class="js-clear-todos"><h3>Clear Checked</h3></a>
+    </div>
+
+  </center>
+
+  <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
+  <script src="store.js"></script>
```
Comment (Charlotteis) [inline demo/Public/index.html]: Yay! I'm excited you managed to get store working. I couldn't manage it :(

File: demo/Public/style.css [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+#body {
+    font-family: "Courier New", Courier, monospace;
+    font-size: 100%;
+    background: #faebd7;
+    color: CF5300;
+    width: 600px;
+    height: 300px;
+    align-items: center;
+
+    }
```
Comment (Charlotteis) [inline demo/Public/style.css]: Looks like a few braces are indented when they shouldn't be in this file. All should be like you've done in the `#logo-container` declaration.

**Comment by Charlotteis** [comment]
I know I have added a lot of comments, but please don't be discouraged. They are all minor things that will make your code more robust and accessible in the long term. This is a great app, I'm really excited we have a demo!

**Comment by Charlotteis** [comment]
Looks like the todo items might want a nice `text-align: left` on them so they are aligned nicely :) And then centering with margins to get them back into the middle of the page :)

rocketship-todos-misaligned

File: demo/Public/style.css [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+h3 {
+  color: #CF5300
+  }
+
+h1 {
+  color: #CF5300
+}
+
+h2 {
```
Comment (Charlotteis) [inline demo/Public/style.css]: You don't seem to be using an H2 in your template at all, so this CSS declaration isn't needed.

File: demo/Public/index.html (hunk 7 above)
Comment (Charlotteis) [inline demo/Public/index.html]: I think this should be a button instead of a link. A link takes you places, a button performs an action. Also, I don't think a H2 fits here as it is not acting as a header for content. If you do use an heading however, it should be an H2 else your heading structure is incorrect.

[tota11y](khan.github.io/tota11y/) says the following:

> "This document contains an `<h3>` tag directly following an `<h1>`. In order to maintain a consistent outline of the page for assistive technologies, reduce the gap in the heading level by upgrading this tag to an `<h2>`."

File: demo/Public/index.html [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+  <meta charset="utf-8">
+  <link rel="stylesheet" type="text/css" href="style.css">
+
+</head>
+<body>
+
+  <center>
+
+    <div id="logo-container"><img src="Rocket.png" /></div>
```
Comment (Charlotteis) [inline demo/Public/index.html]: Can we add `alt=""` to the img? 

http://a11yproject.com/posts/alt-text/

File: demo/Public/index.html (hunk 4 above)
Comment (nickcolley) [inline demo/Public/index.html]: I'd also add that where ever possible try not to use IDs as they lead to very specific CSS that's hard to override.

See also: http://csswizardry.com/2011/09/when-using-ids-can-be-a-pain-in-the-class/

File: package.json [hunk 13]
```diff
@@ -41,6 +41,7 @@
     "joi": "^6.6.1",
     "pouchdb-hoodie-store": "^1.0.1",
     "random-string": "^0.1.2",
-    "spawn-pouchdb-server": "^3.0.1"
+    "spawn-pouchdb-server": "^3.0.1",
+    "standard": "^5.1.1"
```
Comment (gr2m) [inline package.json]: please don't add "standard" to "dependencies", it's already in "devDependencies" :)

**Comment by gr2m** [comment]
I'm sorry there was a misunderstanding. You only need to move the contents of the `public` folder of https://github.com/ShellyCoen/rocketship into the `demo` folder of this repository. So https://github.com/ShellyCoen/rocketship/blob/master/public/index.html becomes https://github.com/hoodiehq/hapi-couchdb-store/blob/master/demo/index.html

Makes sense?

**Comment by ShellyCoen** [comment]
MAkes total sense Gregor, my bad! Should be fixed now.

**Comment by gr2m** [comment]
merged via 09f1e73
